include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -87,13 +93,23 @@ typename MatType::elem_type NSGA2::Optimize(
       BaseMatType>();
   RequireDenseFloatingPointType<BaseMatType>();
 
-  // Check if lower bound is a vector of a single dimension.
-  if (lowerBound.n_rows == 1)
-    lowerBound = lowerBound(0, 0) * arma::ones(iterate.n_rows, iterate.n_cols);
+  BaseMatType castedLowerBound;
+  BaseMatType castedUpperBound;
 
-  // Check if upper bound is a vector of a single dimension.
-  if (upperBound.n_rows == 1)
-    upperBound = upperBound(0, 0) * arma::ones(iterate.n_rows, iterate.n_cols);
+  // Check if lower/upper bound is a vector of a single dimension.
+  if (lowerBound.n_elem == 1)
+  {
+    castedLowerBound = lowerBound(0) * BaseMatType(
+        iterate.n_rows, iterate.n_cols, GetFillType<MatType>::ones);
+
+    castedUpperBound = upperBound(0) * BaseMatType(
+        iterate.n_rows, iterate.n_cols, GetFillType<MatType>::ones);
+  }
+  else
+  {
+    castedLowerBound = conv_to<BaseMatType>::from(lowerBound);
+    castedUpperBound = conv_to<BaseMatType>::from(upperBound);
+  }
 
   // Check the dimensions of lowerBound and upperBound.
   assert(lowerBound.n_rows == iterate.n_rows && "The dimensions of "
```

Comment:
Do these checks now need to concern `castedLowerBound` since we are not updating `lowerBound` anymore?  (Should we still update `lowerBound` and `upperBound` as before?)